test/gtest/ucp/test_ucp_request.cc

```diff
@@ -448,47 +527,209 @@ UCS_TEST_P(test_proto_reset, am_eager_multi_bcopy, "ZCOPY_THRESH=inf",
     reset_protocol(AM);
 }
 
-UCS_TEST_P(test_proto_reset, tag_eager_multi_zcopy_to_bcopy, "ZCOPY_THRESH=0",
+UCS_TEST_P(test_proto_reset, tag_eager_multi_zcopy, "ZCOPY_THRESH=0",
            "RNDV_THRESH=inf")
 {
     reset_protocol(TAG);
 }
 
-UCS_TEST_P(test_proto_reset, get_offload_zcopy_to_get_am_bcopy,
-           "ZCOPY_THRESH=0", "RNDV_THRESH=inf")
+UCS_TEST_SKIP_COND_P(test_proto_reset, get_offload_zcopy, no_rma_transport(),
+                     "ZCOPY_THRESH=0", "RNDV_THRESH=inf",
+                     "RMA_ZCOPY_MAX_SEG_SIZE=1024")
 {
     skip_no_pending_rma();
     reset_protocol(RMA_GET);
 }
 
-UCS_TEST_P(test_proto_reset, put_offload_zcopy_to_put_am_bcopy,
-           "ZCOPY_THRESH=0", "RNDV_THRESH=inf")
+UCS_TEST_P(test_proto_reset, put_offload_zcopy, "ZCOPY_THRESH=0",
+           "RNDV_THRESH=inf", "RMA_ZCOPY_MAX_SEG_SIZE=1024")
 {
     skip_no_pending_rma();
     reset_protocol(RMA_PUT);
 }
 
-UCS_TEST_P(test_proto_reset, stream_multi_zcopy_to_bcopy, "ZCOPY_THRESH=0",
+UCS_TEST_P(test_proto_reset, stream_multi_zcopy, "ZCOPY_THRESH=0",
            "RNDV_THRESH=inf")
 {
     reset_protocol(STREAM);
 }
 
-UCS_TEST_P(test_proto_reset, rndv_am_zcopy_to_bcopy, "ZCOPY_THRESH=0",
-           "RNDV_THRESH=0", "RNDV_SCHEME=am")
+UCS_TEST_P(test_proto_reset, rndv_am_zcopy, "ZCOPY_THRESH=0", "RNDV_THRESH=0",
+           "RNDV_SCHEME=am")
 {
     reset_protocol(TAG);
 }
 
-UCS_TEST_P(test_proto_reset, am_eager_multi_zcopy_to_bcopy, "ZCOPY_THRESH=0",
+UCS_TEST_P(test_proto_reset, am_eager_multi_zcopy, "ZCOPY_THRESH=0",
            "RNDV_THRESH=inf")
 {
     reset_protocol(AM);
 }
 
-UCS_TEST_P(test_proto_reset, rndv_put, "RNDV_THRESH=0", "RNDV_SCHEME=put_zcopy")
+UCS_TEST_P(test_proto_reset, rndv_put, "RNDV_THRESH=0", "RNDV_SCHEME=put_zcopy",
+           "RMA_ZCOPY_MAX_SEG_SIZE=1024")
 {
+    skip_no_pending_rma();
     reset_protocol(TAG);
 }
 
 UCP_INSTANTIATE_TEST_CASE(test_proto_reset)
+
+/* The following tests require ENABLE_DEBUG_DATA flag in order to access
+ * req->recv.proto_rndv_request, which is only present with this flag. */
+#if ENABLE_DEBUG_DATA
+class test_proto_reset_rndv_get : public test_proto_reset {
+protected:
+    void wait_and_restart(const std::vector<request_pair_t> &pairs) override
+    {
+        wait_for_condition(pairs, [](const request_pair_t &pair) {
+            if (pair.rreq->recv.proto_rndv_request == NULL) {
+                return false;
+            }
+
+            const ucp_request_t *rndv_req = pair.rreq->recv.proto_rndv_request;
+
+            return (rndv_req->send.state.dt_iter.offset > 0) &&
+                   (rndv_req->send.state.dt_iter.offset <
+                    rndv_req->send.state.dt_iter.length);
+        });
+
+        restart(receiver().ep());
+    }
+};
+
+UCS_TEST_SKIP_COND_P(test_proto_reset_rndv_get, rndv_get, no_rma_transport(),
+                     "RNDV_THRESH=0", "RNDV_SCHEME=get_zcopy",
+                     "RMA_ZCOPY_MAX_SEG_SIZE=1024")
+{
+    skip_no_pending_rma();
```

Comment:
can this condition be added to `UCS_TEST_SKIP_COND_P`?

Reply:
need to specify the reason for skip (`UCS_TEST_SKIP_R("RMA transport does not support pending queue");`)

Reply:
seems like entities are not yet initialized in that stage.

Reply:
maybe we can check UCT capabilities in worker->wiface[i], rather than by transport names?

Reply:
you mean regardless to the location of the call?